src/analysis/play-utilities/PlayDecider.cpp

```diff
@@ -6,5 +6,18 @@
 
 namespace rtt::ai::analysis {
 // Maybe plays should have some sort of score based on the world how good they are
-Play PlayDecider::decideBestPlay(world::World *world, world::Field *field, std::vector<Play> validPlays) { return Play(); }
+Play* PlayDecider::decideBestPlay(world::World* world, const world::Field& field, std::vector<Play*> const& validPlays) {
+    int max = -1;
+    if (validPlays.empty()) {
+        std::cerr << "valid plays are empty" << std::endl;
+    }
+    for (auto& play : validPlays) {
```

Comment:
Just wanted to throw this suggestion since haico said he preferred less code:

```cpp
bestPlay = std::max_element(validPlays.begin(), validPlays.end(), [&](auto& elem) {
    return elem->scorePlay(world, field);
}); // already a pointer to it :)
```

Reply:
clear code > less code